src/lib/libwasmfs_fetch.js

```diff
@@ -73,17 +117,36 @@ addToLibrary({
       // read/getSize fetch the data, then forward to the parent class.
       read: async (file, buffer, length, offset) => {
         try {
-          await getFile(file);
-        } catch (response) {
-          return response.status === 404 ? -{{{ cDefs.ENOENT }}} : -{{{ cDefs.EBADF }}};
+          await getFileRange(file, offset || 0, length);
+        } catch (failedResponse) {
+          return failedResponse.status === 404 ? -{{{ cDefs.ENOENT }}} : -{{{ cDefs.EBADF }}};
+        }
+        var fileInfo = wasmFS$JSMemoryRanges[file];
+        var fileData = fileInfo.chunks;
+        var chunkSize = fileInfo.chunkSize;
+        var firstChunk = (offset / chunkSize) | 0;
+        var lastChunk = ((offset+length) / chunkSize) | 0;
+        if (offset + length > lastChunk * chunkSize) {
```

Comment:
Same question as before (perhaps worth a comment?)

Reply:
Good idea, I'll simplify the conditions too.